File: src/snippets/rate-limiting/reference/bun/FixedWindow.ts
import arcjet, { fixedWindow } from "@arcjet/bun";
import { env } from "bun";

const aj = arcjet({
  key: env.ARCJET_KEY!, // Get your site key from https://app.arcjet.com
  rules: [
    fixedWindow({
      mode: "LIVE", // will block requests. Use "DRY_RUN" to log only
      window: "60s", // 60 second fixed window
      max: 100, // allow a maximum of 100 requests
    }),
  ],
});

File: src/snippets/rate-limiting/reference/python-fastapi/FixedWindow.py
import os

from arcjet import Mode, arcjet, fixed_window

aj = arcjet(
    key=os.environ["ARCJET_KEY"],
    rules=[
        fixed_window(
            mode=Mode.LIVE,  # Blocks requests. Use Mode.DRY_RUN to log only
            # Tracked by IP address by default, but this can be customized
            # See https://docs.arcjet.com/fingerprints
            # characteristics=["ip.src"],
            window=60,  # 60 second fixed window
            max=100,  # allow a maximum of 100 requests
        ),
    ],
)
